byzcoin/transaction.go

```diff
@@ -359,6 +359,26 @@ func (instr Instruction) Verify(st ReadOnlyStateTrie, msg []byte) error {
 	return instr.VerifyWithOption(st, msg, nil)
 }
 
+func (instr Instruction) usesForbiddenIdentities() bool {
+	// A synthetic instruction is currently not restricted
+	if instr.synthetic {
+		return false
+	}
+
+	// A genuine instruction cannot use an EVM contract identity
+	evmContractType := darc.Identity{
```

Comment:
For later: we should have constants for the types, not just `1`, `2`, ...

Reply:
Indeed. I resisted the urge to change that in this PR, trying to stay focused ;-) .